### DISCUSSION TITLE
vc-drawings组件，是否支持外部调用绘制api

### DISCUSSION THREAD

**Discussion by Sfhji** [zouyaoji/vue-cesium#516] [Q&A]
想实现一种通过自定义的按钮，去调用插件里面绘制的函数，进行绘制。
![image](https://github.com/zouyaoji/vue-cesium/assets/40118136/53745384-48ee-4979-954a-2afde6d978ce)

**Comment by Sfhji** [comment]
已解决

**Reply by shaopGao** [reply]
请教下老哥是咋解决的

**Reply by Sfhji** [reply]
可以通过vc-drawings的插槽 然后加一个v-show 隐藏dom。然后调用事件就行

**Reply by IKE1001** [reply]
哥哥 咋隐藏呢  可不可以贴个代码 瞅一瞅

**Comment by shaopGao** [comment]
老哥，我能用其他按钮调用相同的功能，但我不知道咋把原来的按钮隐藏起来，能教一教吗

**Reply by zouyaoji** [reply]
用插槽vc-drawings的插槽。在线文档里面就有啊。

https://zouyaoji.top/vue-cesium/#/zh-CN/component/analyses/vc-drawings

![image](https://github.com/zouyaoji/vue-cesium/assets/24430116/cc47ef29-986b-4524-b253-62d311bfc2a8)

**Comment by Sfhji** [comment]
我在这里加了一个v-show="false"


然后定义了一个 ref 进行调用&nbsp;


吴鹏飞
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2024年8月6日(星期二) 下午5:28
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; "State ***@***.***&gt;; 
主题: Re: [zouyaoji/vue-cesium] vc-drawings组件，是否支持外部调用绘制api (Discussion #516)














 
哥哥 咋隐藏呢  可不可以贴个代码 瞅一瞅

**Reply by IKE1001** [reply]
懂了  感谢🙏
